### PR TITLE
Add isCompiletimeEvaluable and isRuntimeEvaluable methods to the Value class

### PR DESCRIPTION
**Pull request by lucaswoj**
Not sure if these are the right names for the methods but I think this functionality will be important.

### CODE REVIEW

**Comment by lucaswoj** [comment]
Bad Lucas. This needs unit tests.

**Comment by lucaswoj** [comment]
Maybe we should just say that everything is compiletime evaluable and add a single `isSynthesizable` method. What do you think, @MurphEngineer?

EDIT: No, because top level io ports are not compiletime evaluable.

**Comment by mtrberzi** [comment]
Right, as you said, top-level I/O is not compile-time eval-able. So I would stay with "isCompileTimeEvaluable()" and change the other to "isSynthesizable()", which is a better description of the condition we're trying to check.

**Comment by lucaswoj** [comment]
Yes. We should refine our terminology. I'll start a glossary page on the wiki

**Comment by mtrberzi** [comment]
Will merge this one soon unless there's anything else you want to add, this looks easy enough

**Comment by lucaswoj** [comment]
Ok. Let me just swap `isRuntimeEvaluable` with `isSynthesizable` and throw up some unit tests.

File: src/main/java/org/whdl/frontend/syntaxtree/Value.java [hunk 1]
```diff
@@ -9,4 +9,15 @@
    * exception.
    */
   public abstract void verify();
+  
+  /*
+   * Returns true if this value can be evaulated at compiletime.
```
Comment (TysonAndre) [inline src/main/java/org/whdl/frontend/syntaxtree/Value.java:14]: evaluated

File: src/main/java/org/whdl/frontend/syntaxtree/Value.java [hunk 2]
```diff
@@ -9,4 +9,15 @@
...
+   * Either this or isRuntimeEvaluable or both must return true.
```
Comment (TysonAndre) [inline src/main/java/org/whdl/frontend/syntaxtree/Value.java]: there's no function called isRuntimeEvaluable. Did you mean isSynthesizable?

File: src/main/java/org/whdl/frontend/syntaxtree/Value.java [hunk 3]
```diff
@@ -9,4 +9,15 @@
...
+   */
+  public abstract boolean isCompiletimeEvaluable();
+  
+  /*
+   * Returns true if this value is synthesizable.
```
Comment (TysonAndre) [inline src/main/java/org/whdl/frontend/syntaxtree/Value.java]: This description is begging the question for `isSynthesizable`. What does synthesizable mean?

**Comment by TysonAndre** [comment]
apart from that seems fine

**Comment by mtrberzi** [comment]
Looks fine; go ahead and resolve conflicts and I will merge.

**Comment by lucaswoj** [comment]
Done!

**Pull request merged**